src/extension/coqPilot.ts

```diff
@@ -98,30 +104,42 @@ export class CoqPilot {
         const cursorPosition = editor.selection.active;
         this.performSpecificCompletionsWithProgress(
             (hole) => positionInRange(cursorPosition, hole.range),
-            editor
+            editor,
+            this.abortController.signal
         );
     }
 
     async performCompletionInSelection(editor: TextEditor) {
         const selection = editor.selection;
         this.performSpecificCompletionsWithProgress(
             (hole) => selection.contains(toVSCodePosition(hole.range.start)),
-            editor
+            editor,
+            this.abortController.signal
         );
     }
 
     async performCompletionForAllAdmits(editor: TextEditor) {
-        this.performSpecificCompletionsWithProgress((_hole) => true, editor);
+        this.performSpecificCompletionsWithProgress(
+            (_hole) => true, 
+            editor, 
+            this.abortController.signal
+        );
+    }
+
+    async toggleCurrentSession() {
```

Comment:
I think it's better to name this method more clearly. For example, something like "abortSession" / "abortRequest" / "terminateSession" / etc. While "toggle" usually means you switch between two states (in both directions)